### ISSUE TITLE
Support fragments on Unions and Interfaces

### ISSUE THREAD

**Issue by danbruder** [Youshido/GraphQL#74]
@fubhy's comments in gitter: 

Fragments currently don't work with interfaces or union types... only normal types or composite types.

E.g. this doesn't work:
```
query {
  somethingThatReturnsAUnion(foo: "bar") {
     ... on Something { // Something is a valid type for this union!
        id
     }
  }
}
```

That is, because we currently only load the inner type (resolve any compose types) and then compare the name of that type with the name of the fragment. In case of unions/interfaces however, we need to also compare it with any of the possible types of the interface/union.
And even possible further nested types.

Will add failing test case.

**Comment by viniychuk** [comment]
Unions work now, have to add Interfaces support

**Comment by viniychuk** [comment]
fixed by @fubhy in the v1.4.1.1
